src/v2/Utils/auth.ts

```diff
@@ -170,3 +173,41 @@ export const logout = async () => {
   const err = await response.json()
   return Promise.reject(new Error(err.error))
 }
+
+export enum Status {
+  LoggedIn,
+  LoggedOut,
+  Invalid,
+}
+
+/**
+ * Ensures authentication is still valid.
+ *
+ * If user is logged out; `Promise` resolves to `Status.LoggedOut`
+ * If user is logged in; `Promise` resolves to `Status.LoggedIn`
+ * If user is logged in with invalid authentication (401); `Promise` resolves to `Status.Invalid`
+ * All other status codes will also resolve to `Status.LoggedIn` because we don't know
+ * whether or not the authentication is valid (error could be something else)
+ */
+export const validate = async (): Promise<Status> => {
+  const currentUser = getENV("CURRENT_USER")
+
+  if (!currentUser) return Status.LoggedOut
+
+  const response = await fetch(`${getENV("API_URL")}/api/v1/me`, {
+    credentials: "same-origin",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+      "X-Requested-With": "XMLHttpRequest",
+      "X-XAPP-TOKEN": getENV("ARTSY_XAPP_TOKEN"),
+      "X-ACCESS-TOKEN": currentUser.accessToken,
+    },
+  })
```

Comment:
Not trying to be difficult here but I'm genuinely confused about when `fetch` should be used - especially given this comment:

> yah at this point [code that uses fetch] should be considered something of a code smell

https://artsy.slack.com/archives/C01ADJNCS5D/p1638998715486800?thread_ts=1638997191.482800&cid=C01ADJNCS5D

I mean I actually find this code quite readable and love it but it conflicts with that statement from @damassi - how do yall think through these issues? 🤔 

thanks!

Reply:
By code-smell I meant: in v2, we get 99% of all data need from MP. If for some reason we're _not_ using MP, there should be an exceptional reason to promote that use-case to the 1%.

Reply:
`fetch` should be used if you need to make some request outside of GraphQL. Here I could just use GraphQL, as @damassi points out; and avoid the header set up.